### PR TITLE
Add FromLiteralParamter as ScalarImplementation dependency

### PR DESCRIPTION
**Pull request by fiedukow**
This is precondition to get some work with `varchar(x)` done and possibly to simplify `decimal` implementation.

### CODE REVIEW

**Comment by fiedukow** [comment]
@cberner Can you review this - it's based (and possible in such simple way) on last refactoring (https://github.com/prestodb/presto/pull/5377).

File: presto-main/src/main/java/com/facebook/presto/type/FromLiteralParameter.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package com.facebook.presto.type;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.PARAMETER})
+public @interface FromLiteralParameter
```
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/type/FromLiteralParameter.java]: I think this should be named `LiteralParameter` to match the declarative naming of other annotations

**Comment by cberner** [comment]
Can you include one of the functions that requires this in this PR? I didn't work on any of the bounded varchar functions, and it's not immediately clear to me where this would be needed.

**Comment by fiedukow** [comment]
This is now part of https://github.com/prestodb/presto/pull/5501

**Pull request closed without merging**